File: src/CorporateMessengerWithAgile/Application/Dto/Summary/SprintSummaryDto.cs
namespace Application.Dto.Summary
{
    public class SprintSummaryDto : BaseDto
    {
        public Guid TeamId { get; set; }
        public DateTime DateStart { get; set; }
        public DateTime DateEnd { get; set; }
        public IReadOnlyList<Guid> TaskItemIds { get; set; } = null!;
    }
}

File: src/CorporateMessengerWithAgile/ReactApp/reactapp.client/src/models/entity/SprintSummaryDto.ts
// src/models/entity/SprintSummaryDto.ts
import { z } from 'zod';
import { BaseDtoSchema } from './BaseDto';
import { GuidSchema } from '../Guid';

export const SprintSummaryDtoSchema = BaseDtoSchema.extend({
    teamId: GuidSchema,
    dateStart: z.coerce.date(),
    dateEnd: z.coerce.date(),
    taskItemIds: z.array(GuidSchema).optional(),
});

export type SprintSummaryDto = z.infer<typeof SprintSummaryDtoSchema>;
